`test_ros2trace.test_trace.TestROS2TraceCLI.test_runtime_disable` is flaky

### Generated by Generative AI

_No response_

### Operating System:

Ubuntu

### ROS version or commit hash:

Rolling

### RMW implementation (if applicable):

_No response_

### RMW Configuration (if applicable):

(Should probably not affect this)

### Client library (if applicable):

_No response_

### 'ros2 doctor --report' output

N/A

### Steps to reproduce issue

The test seems to pass fine locally and in the original ci.ros2.org CI run in #185.

### Expected behavior

Test passes.

### Actual behavior

Test fails: https://ci.ros2.org/view/nightly/job/nightly_linux_debug/3511/testReport/junit/test_ros2trace.test_ros2trace.test.test_ros2trace.test_trace/TestROS2TraceCLI/test_runtime_disable/

```
test/test_ros2trace/test_trace.py:681: in test_runtime_disable
    self.assertTraceNotContains(os.path.join(tmpdir, session_name), DEFAULT_EVENTS_ROS)
test/test_ros2trace/test_trace.py:126: in assertTraceNotContains
    self.assertNotIn(
E   AssertionError: 'ros2:rcl_subscription_init' unexpectedly found in ['ros2:rcl_init', 'ros2:rcl_node_init', 'ros2:rmw_publisher_init', 'ros2:rcl_publisher_init', 'ros2:rclcpp_publish', 'ros2:rclcpp_intra_publish', 'ros2:rcl_publish', 'ros2:rmw_publish', 'ros2:rmw_subscription_init', 'ros2:rcl_subscription_init', 'ros2:rclcpp_subscription_init', 'ros2:rclcpp_subscription_callback_added', 'ros2:rmw_take', 'ros2:rcl_take', 'ros2:rclcpp_take', 'ros2:rcl_service_init', 'ros2:rclcpp_service_callback_added', 'ros2:rmw_take_request', 'ros2:rmw_send_response', 'ros2:rmw_client_init', 'ros2:rcl_client_init', 'ros2:rmw_send_request', 'ros2:rmw_take_response', 'ros2:rcl_timer_init', 'ros2:rclcpp_timer_callback_added', 'ros2:rclcpp_timer_link_node', 'ros2:rclcpp_callback_register', 'ros2:callback_start', 'ros2:callback_end', 'ros2:rcl_lifecycle_state_machine_init', 'ros2:rcl_lifecycle_transition', 'ros2:rclcpp_executor_get_next_ready', 'ros2:rclcpp_executor_wait_for_work', 'ros2:rclcpp_executor_execute', 'ros2:rclcpp_ipb_to_subscription', 'ros2:rclcpp_buffer_to_ipb', 'ros2:rclcpp_construct_ring_buffer', 'ros2:rclcpp_ring_buffer_enqueue', 'ros2:rclcpp_ring_buffer_dequeue', 'ros2:rclcpp_ring_buffer_clear'] : ros2:rcl_subscription_init found in events: [{'_name': 'lttng_ust_tracef:event', '_timestamp': 1754300698423549398, 'msg': 'TRACETOOLS_TEST_TRACE_TEST_ID=test_runtime_disable#412142', 'cpu_id': 0, 'procname': 'test_ping', 'vtid': 414700, '_msg_length': 57, 'vpid': 414700}, {'_name': 'lttng_ust_tracef:event', '_timestamp': 1754300698436208245, 'msg': 'TRACETOOLS_TEST_TRACE_TEST_ID=test_runtime_disable#412142', 'cpu_id': 3, 'procname': 'test_pong', 'vtid': 414701, '_msg_length': 57, 'vpid': 414701}, {'_name': 'ros2:rcl_subscription_init', '_timestamp': 1754300699004733100, 'vtid': 414777, 'topic_name': '/test_time_09_44_58/test/message/Constants', 'queue_depth': 1, 'rmw_subscription_handle': 109095935312512, 'procname': 'test_subscriber', 'vpid': 414777, 'node_handle': 109095922515488, 'cpu_id': 3, 'subscription_handle': 109095935425872}]
```

### Additional information

This might be due to parallel testing in nightly CI. The output in the failed assertion above shows unexpected trace events with a `/test_time_09_44_58/test/message/Constants` topic, which obviously comes from another test.

--- a/test_ros2trace/test/test_ros2trace/test_trace.py
+++ b/test_ros2trace/test/test_ros2trace/test_trace.py
@@ -113,19 +113,23 @@ class TestROS2TraceCLI(unittest.TestCase):
         self.assertFalse(os.path.isdir(trace_dir), f'trace directory exists: {trace_dir}')
 
     def assertTraceNotContains(self, trace_dir: str, unexpected_event_names: List[str]) -> None:
+        self.assertNotIn(
+            TRACE_TEST_ID_TP_NAME, unexpected_event_names, 'process marker event is required')
         self.assertTraceExist(trace_dir)
         from tracetools_read.trace import get_trace_events
         events = get_trace_events(trace_dir)
         trace_test_events = get_corresponding_trace_test_events(events, self.trace_test_id)
+        # The events from the marked processes should contain process marker event(s)
         self.assertGreater(
             len(trace_test_events),
             0,
             f'no matching trace test events found: {events}')
-        for event in events:
+        # but not the unexpected events
+        for event in trace_test_events:
             event_name = get_event_name(event)
             self.assertNotIn(
                 event_name, unexpected_event_names,
-                f'{event_name} found in events: {events}'
+                f'{event_name} found in events: {trace_test_events}'
             )
 
     def assertTraceContains(

--- a/tracetools_test/tracetools_test/mark_process.py
+++ b/tracetools_test/tracetools_test/mark_process.py
@@ -65,7 +65,7 @@ def get_corresponding_trace_test_events(
 
     :param events: all of the trace events
     :param trace_test_id: the trace test ID
-    :return: the relevant trace events corresponding to the given test
+    :return: the relevant trace events corresponding to the given test, including the marker event
     """
     expected_event_msg = _get_trace_test_id_event_value(trace_test_id)
     events_id = get_events_with_name(TRACE_TEST_ID_TP_NAME, events)

